campaign-launcher/interfaceV2/src/utils/index.ts

```diff
@@ -53,11 +58,111 @@ export const getExplorerUrl = (chainId: ChainId, address: string): string => {
 };
 
 export const formatTokenAmount = (amount: string | number, decimals = 18): string | number => {
-  const formattedAmount = +amount / Math.pow(10, decimals);
-  
-  if (formattedAmount >= 1000) {
-    return Math.round(formattedAmount);
+  const bigIntAmount = BigInt(amount);
+  const amountString = formatUnits(bigIntAmount, decimals);
+  const amountNumber = Number(amountString);
+
+  if (!Number.isFinite(amountNumber)) return amountString;
+
+  if (amountNumber >= 1000) {
+    return Math.round(amountNumber);
   } else {
-    return parseFloat(formattedAmount.toFixed(3));
+    return parseFloat(amountNumber.toFixed(4));
+  }
+};
+
+export const mapStatusToColor = (status: Campaign['status'], startDate: string, endDate: string) => {
+  const theme = useTheme();
+  const today = new Date().toISOString();
+
+  switch (status) {
+    case 'active':
+      if (today < startDate) {
+        return theme.palette.warning.main;
+      } else if (today > endDate) {
+        return theme.palette.error.main;
+      } else {
+        return theme.palette.success.main;
+      }
+    case 'cancelled':
+      return theme.palette.primary.main;
+    case 'completed':
+      return theme.palette.secondary.main;
+    default:
+      return theme.palette.primary.main;
+  }
+};
+
+export const getChainIcon = (chainId: ChainId) => {
+  return CHAIN_ICONS[chainId] || null;
+};
+
+export const getNetworkName = (chainId: ChainId): string | null => {
+  return NETWORKS[chainId]?.title || null;
+};
+
+
+export const isCampaignDetails = (obj: unknown): obj is CampaignDetails => {
+  if (!obj || typeof obj !== 'object') return false;
+  
+  const requiredCampaignFields = [
+    'id', 'chain_id', 'address', 'exchange_name', 'trading_pair',
+    'start_date', 'end_date', 'fund_amount', 'fund_token', 
+    'fund_token_symbol', 'fund_token_decimals', 'status',
+    'escrow_status', 'launcher', 'exchange_oracle', 
+    'recording_oracle', 'reputation_oracle', 'amount_paid', 'daily_paid_amounts'
+  ];
+  
+  for (const field of requiredCampaignFields) {
+    if (!(field in obj)) return false;
   }
+  
+  return true;
 };
+
+export const constructCampaignDetails = (chainId: ChainId, address: string, data: EscrowCreateDto) => {
+  const tokenDecimals = data.fund_token === 'hmt' ? 18 : 6;
```

Comment:
`decimals` is not always `6` for other tokens, you should pass its value from the escrow creation process (you already fetch it there using `decimals()`)